File: pay/src/main/kotlin/com/truenine/component/pay/properties/WeChatProperties.java
package com.truenine.component.pay.properties;


import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Data
@ConfigurationProperties(prefix = "component.pay.wechat")
public class WeChatProperties {

  private String merchantId;
  private String privateKeyPath;
  private String merchantSerialNumber;
  private String certPaths;

  /**
   * appId 小程序 Id
   */
  private String appId;
  /**
   * 异步通知 Id
   */
  private String notifyUrl;

}

File: security/oauth2/src/main/kotlin/io/github/truenine/composeserver/security/oauth2/properties/WechatProperties.kt
package io.github.truenine.composeserver.security.oauth2.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.NestedConfigurationProperty

private const val PREFIX = "compose.security.oauth2.wechat"

@ConfigurationProperties(prefix = PREFIX) data class WechatProperties(@NestedConfigurationProperty var wxpa: WxpaProperties = WxpaProperties())
